yarn-project/archiver/src/archiver/config.ts

```diff
@@ -37,5 +49,7 @@ export function getConfigEnvVars(): ArchiverConfig {
     unverifiedDataEmitterContract: UNVERIFIED_DATA_EMITTER_ADDRESS
       ? EthAddress.fromString(UNVERIFIED_DATA_EMITTER_ADDRESS)
       : EthAddress.ZERO,
+    searchStartBlock: SEARCH_START_BLOCK ? +SEARCH_START_BLOCK : 0,
+    apiKey: API_KEY || '',
```

Comment:
I'd avoid using empty strings to signal empty values. Let's change `apiKey` to be `string | undefined` if it could be unset.